lib/src/kbasesearchengine/main/AccessGroupNarrativeInfoProvider.java

```diff
@@ -42,15 +42,23 @@ public AccessGroupNarrativeInfoProvider(final WorkspaceEventHandler wsHandler) {
     @Override
     public NarrativeInfo findNarrativeInfo(final Long accessGroupID) {
         Tuple9 <Long, String, String, String, Long, String, String,
-                String, Map<String,String>> wsInfo;
+                String, Map<String,String>> wsInfo = null;
+
+            try {
+                wsInfo = wsHandler.getWorkspaceInfo(accessGroupID);
+            } catch (IOException e) {
+                if (e.getMessage().toLowerCase().contains("is deleted") ||
+                        e.getMessage().toLowerCase().contains("has been deleted")) {
+                    wsInfo = null;
```

Comment:
Correct me if I'm wrong, but I think setting wsInfo to null again is redundant, since its set to null when its first defined..